### ISSUE TITLE
ConstructorParameterNaming false positive

### ISSUE THREAD

**Issue by fluxxion82** [detekt/detekt#4360]
## Expected Behavior
New private constructor parameter should not be flagged.

## Observed Behavior
New private constructor parameter is being flagged as 'ConstructorParameterNaming' by detekt. If the parameter is made public, then it is not flagged by detekt despite having the same regex.

## Steps to Reproduce
Have project with baseline. Add new private constructor parameter.

## Context
Our ci is failing because detekt is making false positives.

## Your Environment
* Version of detekt used: 1.19.0-RC2 & 1.19.0
* Version of Gradle used (if applicable): 4.2.2
* Gradle scan link (add `--scan` option when running the gradle task): https://gradle.com/s/z4pjk2zouvb5i
* Operating System and version: MacOS Big Sur 11.6
* Link to your project (if it's a public repository): N/A

**Comment by BraisGabin** [comment]
Could you provide a snippet code to reproduce your issue? I think that I didn't understand completely your use case.

**Comment by fluxxion82** [comment]
So I have an existing project and we've setup detekt and everything. Most of the configs are the default. When we update a class and add a new private constructor parameter, then detekt is failing.

For example, we have an existing class:
```
class ExampleViewModel(
    private val paramOne: SomeObjectOne,
    private val paramTwo: SomeObjectTwo
) : BaseViewModel
```
Then we add a new private constructor parameter:

```
class ExampleViewModel(
    private val paramOne: SomeObjectOne,
    private val paramTwo: SomeObjectTwo,
    private val paramNew: NewObjectAdded
) : BaseViewModel
```
and this update will fail the detekt check. If we remove the `private` modifier, then it passes the detekt check. Here is our config for the constructor parameter naming:

```
  ConstructorParameterNaming:
    active: true
    excludes: ['**/test/**', '**/androidTest/**']
    parameterPattern: '[a-z][A-Za-z0-9]*'
    privateParameterPattern: '[A-Z$][a-zA-Z0-9]*'
    excludeClassPattern: '$^'
    ignoreOverridden: true
```

Not sure if it matters, but I think these classes have entries (other than ConstructorParameterNaming) in the baseline.

Let me know if you need more info.

**Comment by severn-everett** [comment]
The first `ExampleViewModel` example (i.e. with two constructor params) actually fails Detekt for me given the modified `privateParameterPattern` regex (i.e. `[A-Z$][a-zA-Z0-9]*`). What are the other entries in the baseline, and what does `BaseViewModel` look like?

**Comment by BraisGabin** [comment]
All your params doesn't follow your pattern. Maybe what you should change is your configuration.

The default configuration is:
```yaml
privateParameterPattern: '[a-z][A-Za-z0-9]*'
```

**Comment by fluxxion82** [comment]
So that example was just some code I wrote out in the moment and not my actual code. Regardless of the pattern, private and non private constructor params have the same config, and only the private param is getting flagged. I've tried updating my config pattern but that didn't help either.

**Comment by BraisGabin** [comment]
So, can you provide a real reproducer? Or at least your real configuration? Otherwise we can't help you. I can't reproduce your behaviour on my end.

**Comment by cortinico** [comment]
> Regardless of the pattern, private and non private constructor params have the same config, and only the private param is getting flagged. I've tried updating my config pattern but that didn't help either.

I think we really need a repro project. I'm afraid the issue you're having is a misconfiguration of Detekt and not a bug in the rule.

**Comment by fluxxion82** [comment]
I did give you my configuration. I'll see about putting a repo project together

**Comment by cortinico** [comment]
> I did give you my configuration

Thanks for that, but if you could take the time to set up a reproducer, we would be extremely grateful. We're receiving a high amount of issues/PRs and our time is extremely limited. Issues that contains repros that are easy to checkout and verify are more likely to get fixed 🙏 

Providing a reproducer also sometime turns out to be a great exercise for the reporter. Some of the issue gets magically solved while creating a reproducer just because of a mis-configuration on the reporter's side.

**Comment by BraisGabin** [comment]
> I did give you my configuration. I'll see about putting a repo project together


> Regardless of the pattern, private and non private constructor params have the same config

Sorry but this two sentences doesn't match. If you provided your real configuration the problem is in your configuration as we pointed out. 

If in your configuration the private and non private params have the same config the configuration that you provided us is not your real configuration. The snippet that you posted has different config for those two type of params.

**Comment by BraisGabin** [comment]
Closing because we can't reproduce the behaviour.
